std/src/main/scala/algebra/std/array.scala

```diff
@@ -0,0 +1,42 @@
+package algebra
+package std
+
+import scala.{specialized ⇒ sp}
+
+package object array extends ArrayInstances
+
+trait ArrayInstances {
+  implicit def arrayEq[@sp A: Eq]: Eq[Array[A]] = new ArrayEq[A]
```

Comment:
I don't really like the fact that an array comparison can return values other than 0, -1, +1. Want me to change that? Also, want me to squash the commits once we have something that we're happy with?